src/core/header.rs

```diff
@@ -0,0 +1,58 @@
+use crate::core::digest::Digests;
+use crate::core::{Hash, U256};
+
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize, PartialEq, Eq)]
+pub struct Header<H: Hash, N: Copy + Into<U256> + TryFrom<U256>> {
+    /// the parent hash
+    pub parent_hash: H::Output,
+    /// block number or block height
+    pub number: N,
+    /// the root of state tired merkle tree
+    pub state_root: H::Output,
+    /// the merkle root of current block  
+    pub block_root: H::Output,
+    /// aux data for consensus
+    pub digests: Digests,
+}
+
+impl<H: Hash, N: Copy + Into<U256> + TryFrom<U256>> Header<H, N> {
+    fn new(
+        number: N,
+        block_root: H::Output,
+        state_root: H::Output,
+        parent_hash: H::Output,
+        digests: Digests,
+    ) -> Self {
+        Self {
+            parent_hash,
+            number,
+            state_root,
+            block_root,
+            digests,
+        }
+    }
+
+    fn number(&self) -> &N {
```

Comment:
Hmm, these fields are `pub`lic.  Do we need those getter functions? Is it a best practice in Rust?

Reply:
At first, I wanted to make some bounds as the trait of header. Now the getters has been removed. The header has been re-exported on the path `core/mod.rs`
```
pub type BlockNumU64 = u64;
pub type Sha3_256 = crate::core::hash::Sha3Hasher;
pub type Header = crate::core::header::Header<Sha3_256, BlockNumU64>;
pub type Block = crate::core::blocks::Block<Header>;
```
So, It can be direct, simple and practical. Some other crate has no more generics when `block` in use.